### ISSUE TITLE
Undefined references when building

### ISSUE THREAD

**Issue by Lastin** [buggins/dlangui#139]
Hi, I tried building hello world as instructed, but I get following error.
```
Building: dlangui-monod-linux (Debug|x64)
Performing main compilation...
Build complete -- 0 errors, 0 warnings

Building: helloworld-monod-linux (Debug|x64)
Performing main compilation...

Building: dlangui-monod-linux (Debug|x64)
Performing main compilation...
Build complete -- 0 errors, 0 warnings
Current dictionary: /home/maksym/Desktop/dlangui/examples/helloworld
dmd -m64 -debug -version=USE_SDL -version=USE_OPENGL -version=USE_FREETYPE -gc "src/helloworld.d" "/home/maksym/Desktop/dlangui/bin/Debug/libdlangui-monod-linux.a" "-I/usr/include/dmd" "-I/home/maksym/Desktop/dlangui/src" "-I/home/maksym/Desktop/dlangui/3rdparty" "-I/home/maksym/Desktop/dlangui/deps/gl3n" "-I/home/maksym/Desktop/dlangui/deps/DerelictSDL2/source" "-I/home/maksym/Desktop/dlangui/deps/DerelictFT/source" "-I/home/maksym/Desktop/dlangui/deps/DerelictGL3/source" "-I/home/maksym/Desktop/dlangui/deps/DerelictUtil/source" "-I/home/maksym/Desktop/dlangui" "-odobj/Debug" "-of/home/maksym/Desktop/dlangui/examples/helloworld/bin/Debug/helloworld-monod-linux" -w -vcolumns
/home/maksym/Desktop/dlangui/bin/Debug/libdlangui-monod-linux.a(sharedlib_2f92_4ee.o): In function `_D8derelict4util9sharedlib13LoadSharedLibFAyaZPv':
/home/maksym/Desktop/dlangui/deps/DerelictUtil/source/derelict/util/sharedlib.d:43: undefined reference to `dlopen'
/home/maksym/Desktop/dlangui/bin/Debug/libdlangui-monod-linux.a(sharedlib_2f93_5d1.o): In function `_D8derelict4util9sharedlib15UnloadSharedLibFPvZv':
/home/maksym/Desktop/dlangui/deps/DerelictUtil/source/derelict/util/sharedlib.d:47: undefined reference to `dlclose'
/home/maksym/Desktop/dlangui/bin/Debug/libdlangui-monod-linux.a(sharedlib_2f94_396.o): In function `_D8derelict4util9sharedlib9GetSymbolFPvAyaZPv':
/home/maksym/Desktop/dlangui/deps/DerelictUtil/source/derelict/util/sharedlib.d:51: undefined reference to `dlsym'
/home/maksym/Desktop/dlangui/bin/Debug/libdlangui-monod-linux.a(sharedlib_2f95_463.o): In function `_D8derelict4util9sharedlib11GetErrorStrFZAya':
/home/maksym/Desktop/dlangui/deps/DerelictUtil/source/derelict/util/sharedlib.d--- errorlevel 1
:55: undefined reference to `dlerror'
collect2: error: ld returned 1 exit status
Exit code 1
Build complete -- 1 error, 0 warnings

---------------------- Done ----------------------

Build successful! - No new linkage needed
Build successful! - No new linkage needed
Build: 1 error, 0 warnings
```
I'm using monodevelop on Ubuntu 14.04.
Am I doing something wrong?

**Comment by buggins** [comment]
does DUB build work ok?

**Comment by Lastin** [comment]
Yes it builds and runs just fine.

**Comment by buggins** [comment]
It worked for me under Ubuntu 15.10
Is your monodevelop up to date?
Possible fix could be adding of -L-ldl or -ldl to linker settings in Project Properties

**Comment by Lastin** [comment]
Monodevelop version 5.10
I have tried with -L-ldl and -ldl but without success.
I will just install Ubuntu 15.10 if you say it works. Thank you for your time.

**Comment by Lastin** [comment]
I have actually been using newer monodevelop on 14.04 (version 5.10) than default available on 15.04 (5.5). I am not sure if that's the reason, but I do confirm that it works for me on Ubuntu 15.10

**Comment by buggins** [comment]
Closing bug so far
